File: 1716-calculate-money-in-leetcode-bank/1716-calculate-money-in-leetcode-bank.cpp
class Solution {
public:
    int dp[7]={ 1,2,3,4,5,6,7};
    int totalMoney(int n) {
        
        int ans=0;
        for( int i=0;i<n;i++){
            ans+=dp[i%7];
            dp[i%7]++;
        }
        return ans;
    }
};

File: 1716-calculate-money-in-leetcode-bank/1716-calculate-money-in-leetcode-bank.java
class Solution {
    int[] dp = {1, 2, 3, 4, 5, 6, 7};

    public int totalMoney(int n) {
        int ans = 0;
        for (int i = 0; i < n; i++) {
            ans += dp[i % 7];
            dp[i % 7]++;
        }
        return ans;
    }
}